proof-systems/src/darlin/poly-commit/src/constraints/tests.rs

```diff
@@ -0,0 +1,584 @@
+    negative_type: Option<NegativeTestType>,
+}
+
+fn value_for_alloc<F: Field, R: RngCore>(
+    value: &F,
+    negative_type: &Option<NegativeTestType>,
+    rng: &mut R,
+) -> F {
+    if let Some(NegativeTestType::Values) = negative_type {
+        F::rand(rng)
+    } else {
+        value.clone()
+    }
+}
+
+fn commitment_for_alloc<G: Group, PC: PolynomialCommitment<G>>(
+    commitment: &PC::Commitment,
+    negative_type: &Option<NegativeTestType>,
+) -> PC::Commitment {
+    if let Some(NegativeTestType::Commitments) = negative_type {
+        commitment.clone().double()
+    } else {
+        commitment.clone()
+    }
+}
+
+fn test_succinct_verify_template<
+    ConstraintF: PrimeField,
+    G: Group<BaseField = ConstraintF>,
+    PC: PolynomialCommitment<G>,
+    PCG: PolynomialCommitmentVerifierGadget<ConstraintF, G, PC>,
+>(
+    test_conf: TestInfo,
+) -> Result<(), PCG::Error> {
+    let rng = &mut thread_rng();
+    for _ in 0..test_conf.num_iters {
+        let max_degree: usize = test_conf.max_degree.unwrap_or(rng.gen_range(1..=256));
+        let supported_degree: usize = test_conf
+            .supported_degree
+            .unwrap_or(rng.gen_range(1..=max_degree));
+        assert!(supported_degree <= max_degree);
+
+        let pp = PC::setup::<Blake2s>(max_degree)?;
+        let (ck, vk) = pp.trim(supported_degree)?;
+        let poly_degree: usize = if test_conf.segmented {
+            rng.gen_range(supported_degree..=10 * supported_degree)
+        } else {
+            rng.gen_range(0..=supported_degree)
+        };
+        let polynomial = Polynomial::rand(poly_degree, rng);
+        let is_hiding: bool = rng.gen();
+
+        let (mut commitment, randomness) = PC::commit(&ck, &polynomial, is_hiding, Some(rng))?;
+
+        let point = G::ScalarField::rand(rng);
+        let mut value = polynomial.evaluate(point);
+        let fs_seed = String::from("TEST_SEED").into_bytes();
+        let mut fs_rng =
+            PC::RandomOracle::from_seed(fs_seed.clone()).map_err(|err| PolyError::from(err))?;
+        let proof = PC::open(
+            &ck,
+            polynomial,
+            point,
+            is_hiding,
+            randomness,
+            &mut fs_rng,
+            Some(rng),
+        )?;
+        // check that verify with primitive succeeds
+        let mut fs_rng =
+            PC::RandomOracle::from_seed(fs_seed.clone()).map_err(|err| PolyError::from(err))?;
+        let v_state = PC::succinct_verify(&vk, &commitment, point, value, &proof, &mut fs_rng)?;
+        if v_state.is_none() {
+            Err(PolyError::FailedSuccinctCheck)?
+        }
+
+        let mut cs = ConstraintSystem::<ConstraintF>::new(SynthesisMode::Debug);
+        match test_conf.negative_type {
+            Some(NegativeTestType::Commitments) => {
+                commitment = commitment * &G::ScalarField::rand(rng)
+            }
+            Some(NegativeTestType::Values) => value = G::ScalarField::rand(rng),
```

Comment:
Why are you doing this directly here ? Aren't you suppose to use (and I see you use them later again) the commitment_for_alloc and value_for_alloc functions ?

Reply:
Thanks, they are a leftover, I will remove them